### PR TITLE
build(other): add gh action for deploying release

### PR DESCRIPTION
**Pull request by waldenraines**
## Description

Use a github action to deploy the build to github pages after a
release is created.

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

## Does this PR fix an open issue?

* [ ] Yes
* [x] No

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Build and Deploy
+on: [release]
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.1 # If you're using actions/checkout@v2 you must set persist-credentials to false in most cases for the deployment to work correctly.
```
Comment (humanumbrella) [inline .github/workflows/main.yml:8]: remove comment

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        with:
+          persist-credentials: false
+
+      - name: Install and Build # This example project is built using npm and outputs the result to the 'build' folder. Replace with the commands required to build your project, or remove this step entirely if your site is pre-built.
```
Comment (humanumbrella) [inline .github/workflows/main.yml:12]: remove comment

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        run: |
+          sudo apt-get install libudev-dev libusb-1.0-0-dev
+          npm install
+          npm run build
+
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@3.7.1
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          BRANCH: gh-pages # The branch the action should deploy to.
```
Comment (humanumbrella) [inline .github/workflows/main.yml:22]: same

**Review by humanumbrella — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 3 above)
Comment (waldenraines) [reply .github/workflows/main.yml:22]: Copy/pasta detected

**Comment by waldenraines** [comment]
Addressed all comments.

**Review by bucko13 — APPROVED** [review]

**Comment by waldenraines** [comment]
Approved by @bucko13 elsewhere.

**Pull request merged**